### PR TITLE
DPE-2189 Miscellaneous updates to get snap ready for stable release

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
1. We have compiled (copy/pasted) package licenses into the repository. This means that the licenses may be stale if changed from the time they were copied into the repo
2. The license field for the snap is missing in snapcraft.yaml
3. Our snap's versions are similar to `8.0.33-0` instead of `8.0.33`
4. We need `sudo` to run `charmed-mysql.mysqlsh` because the script for mysqlsh uses `setpriv`

## Solution
1. Organize licenses/copyright files during stage from the installed package docs to `$SNAP/licenses`
2. Add the license field with value `Apache-2.0`
3. Change versions to strings similar to `8.0.33`
4. Create a [snapcraft forum post](https://forum.snapcraft.io/t/specifying-snap-daemon-as-user-when-running-a-snap-application/36267) asking how to run the application as the `snap_daemon` user

### CODE REVIEW

**Review by delgod — APPROVED** [review]

**Review by carlcsaposs-canonical — APPROVED** [review]

**Review by paulomach — APPROVED** [review]

File: .github/workflows/check_version.yaml
Comment (taurus-forever) [inline .github/workflows/check_version.yaml:1]: Please remove Foresight telemetry (can be separate PR).
Foresight R.I.P:

      - name: 'Foresight: Collect workflow telemetry'
        uses: runforesight/foresight-workflow-kit-action@v1

File: .github/workflows/check_version.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
         run: |
           SNAP_VERSION=$(yq .version snap/snapcraft.yaml)
           APT_VERSION=$(apt info mysql-server | grep Version | cut -c10-)
-          APT_VERSION=${APT_VERSION::-15}
+          APT_VERSION=${APT_VERSION::-17}
```
Comment (taurus-forever) [inline .github/workflows/check_version.yaml]: I think we need to add some love here:
1) we are installing debian package `mysql-server-8.0` but checking version of metapackage `mysql-server`.
2) `apt info` shows version in "apt metadata" and not the version of "installed" debian package.
2) "apt" produces warnings on STDERR: `WARNING: apt does not have a stable CLI interface. Use with caution in scripts.`
3) cutting last 17 symbols is an interesting approach... maybe this is nicer:
```
APT_VERSION=$(dpkg-query --show --showformat="\${source:Upstream-Version}" "mysql-server-8.0")
```

**Review by taurus-forever — CHANGES_REQUESTED** [review]
Nice work done! Superb!
One shell related comment I cannot ignore... meanwhile you have a chance to clean Foresight here as well :-)

File: .github/workflows/check_version.yaml (hunk 1 above)
Comment (shayancanonical) [reply .github/workflows/check_version.yaml]: Discussed offline: `dpkg-query` only works for installed snaps. Updated to use `apt info` on `mysql-server-8.0` (instead of `mysql-server`) in [66559e9](https://github.com/canonical/charmed-mysql-snap/pull/36/commits/66559e9e60c069b092c7ba4b2eaba742f387f70e)

Comment (carlcsaposs-canonical) [reply .github/workflows/check_version.yaml]: re: apt does not have a stable CLI interface

I think you can use `apt-cache show` instead of `apt info` (not sure if that's the exact command; there should be an `apt-get` or `apt-cache` command that corresponds to `apt info`)

Comment (shayancanonical) [reply .github/workflows/check_version.yaml]: Use `apt-cache show`'s latest version entry in the version check (to address apt's unstable cli interface)

**Review by taurus-forever — APPROVED** [review]
Let's move forward! Tnx!
